# SARS-CoV2-Phylogenetics
Phylogenetic Analysis of SARS-CoV2 Sequences, USA!

The objective of this project is to extract, classify and analyze the data of the protein sequence of the SARS-Cov2 from the 1.45GB AllSurfaceGlycoproteinSeqs.fasta file. Using Biopython libraries the ultimate goal is to perform multiple sequence alignments and phylogenetic analysis on sequences grouped by geographic regions as determined by the United States Census Bureau. The multiple sequence alignments and phylogenetic tress that will be obtained will reveal how the virus has evolved due to different kinds of selective pressure and how the virus had spread in different regions based on the actions that have been put in place to mitigate its spread. 

The genome of covid-19 is approximately 29kb in length and consists of ’16 nonstructural proteins (nsps), 4 structural proteins, and 9 accessory proteins’ (2). The spike protein on the surface of the virion is the key to entry to human cells. ).The spike protein contains two major domains/subunits: the S1 and the S2 subunits.The S1 subunit contains several domains, among them, a receptor-binding-domain (RBD). The RBD allows for the binding of the spike protein to the receptors present on host cells (ACE2, TMPRSS2).

![Screen Shot 2022-01-02 at 12 22 57 PM](https://user-images.githubusercontent.com/90015489/147883946-e94aae51-bb49-443c-bd01-5cb90d689784.png)

Methods

Biopython libraries were essential in the parsing and processing of data. The libraries used include SeqIO, AlignIO, PhyloIO, matplotlib and networxx. SeqIO was used in order to process and parse FASTA-formatted files and group them based on the State where the specimen sequence was obtained. Data from each state was cleaned through pattern searching using the regular expressions module, re, in order to obtain the sequence of the Receptor Binding Domain and write a message regarding the amino acid positions where the RBD started and ended. The files were once again cleaned up in order to include only the RBD without the message previously printed. The text was originally removed to save storage space on each file considering  ClustalW’s file size processing limit of 4MB. The last part of the file preparation process was the extraction of only 100 sequences from each state. The reason why only 100 sequences were used was because of the file size limit just mentioned. If the entire sequences from each state were used, the individual and resulting files would surpass 4MB by many orders of magnitude. While extracting the first 100 sequences from each state, each entry was numbered in order to ensure uniqueness when producing a multiple sequence alignment. ClustalW requires each entry description to be unique otherwise the sequences will not align and an error message will indicate that duplicate sequences were found.
The states were grouped by geographic region as defined by the United States census (1). Six regions were used to classify the states: Northeast, Southeast,Southwest, Mountain, Midwest, and Pacific.
Once the states had been grouped by regions, a Multiple Sequence Alignment (MSA) was performed ClustalW installed on the local drive. .dnd and .aln (CLUSTAL alignment format) files were produced. The guide tree file (.dnd) was used in order to create the phylogenetic trees for each group directly on jupyter notebook. Matplotlib was used in order to manipulate the phylogenetic tree figure that was produced to fix the dimensions and text size. The phylogenetic trees that were produced were saved as a .png file and converted to a PDF. 

The phylogenetic trees that were produced were quite large, a sample area of a tree is seen below:
![image](https://user-images.githubusercontent.com/90015489/147884392-aa7abec8-5f13-4ec7-aed1-b331e5db7440.png)


